### PR TITLE
feat(bigtable): Add support for Backup restore to different Instance

### PR DESCRIPTION
**Pull request by quartzmo**
* Add `instance` to `Backup#restore`

closes: #8927

### CODE REVIEW

File: google-cloud-bigtable/acceptance/bigtable/backup_test.rb [hunk 1]
```diff
@@ -107,23 +109,35 @@
...
       _(restore_table.name).must_equal restore_table_id
+      _(restore_table.instance_id).must_equal instance.instance_id
 
       # optimize
       _(restore_job.optimize_table_operation_name).wont_be :nil?
       _(restore_job.optimize_table_operation_name).must_be_kind_of String
       _(restore_job.optimize_table_operation_name).must_include restore_table_id
+
+      # restore to another instance
+      restore_job = backup.restore restore_table_id, instance: instance_2
```
Comment (quartzmo) [inline google-cloud-bigtable/acceptance/bigtable/backup_test.rb:128]: Currently failing here using Ruby CI project:

```
  1) Error:
Google::Cloud::Bigtable::Table::bigtable#test_0001_creates a backup:
Google::Cloud::InvalidArgumentError: 3:Request contains an invalid argument.. debug_error_string:{"created":"@1611009167.366622000","description":"Error received from peer ipv4:142.250.68.74:443","file":"src/core/lib/surface/call.cc","file_line":1063,"grpc_message":"Request contains an invalid argument.","grpc_status":3}
    /Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-bigtable-admin-v2/lib/google/cloud/bigtable/admin/v2/bigtable_table_admin/client.rb:1767:in `rescue in restore_table'
    /Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-bigtable-admin-v2/lib/google/cloud/bigtable/admin/v2/bigtable_table_admin/client.rb:1732:in `restore_table'
    /Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-bigtable/lib/google/cloud/bigtable/service.rb:768:in `restore_table'
    /Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-bigtable/lib/google/cloud/bigtable/backup.rb:386:in `restore'
    /Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-bigtable/acceptance/bigtable/backup_test.rb:128:in `block (2 levels) in <top (required)>'
```

Comment (quartzmo) [reply google-cloud-bigtable/acceptance/bigtable/backup_test.rb:128]: This acceptance test is now passing with the google-cloud-ruby CI project and my dev project.

**Comment by quartzmo** [comment]
This PR is ready for review.

**Review by dazuma — APPROVED** [review]
LGTM

**Comment by kolea2** [comment]
Thanks @quartzmo! For now, can you please change the branch it is targeting from master to a different one (ie `backups-restore-instance` or something similar?) Thanks!

**Review by kolea2 — CHANGES_REQUESTED** [review]
please switch branch, lgtm otherwise

**Comment by quartzmo** [comment]
@kolea2 OK, this PR is now targeting `backups-restore-instance`, which is a new branch off of `master`.

**Review by kolea2 — APPROVED** [review]

**Comment by quartzmo** [comment]
@kolea2 What is the schedule for merging this change into master and releasing? Will you notify via this PR?

**Comment by quartzmo** [comment]
This PR was merged into the topic branch `backups-restore-instance`. It was not merged into `master` branch.

**Comment by quartzmo** [comment]
This was released in [google-cloud-bigtable v2.6.0](https://rubygems.org/gems/google-cloud-bigtable/versions/2.6.0).

**Pull request merged**